pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp

```diff
@@ -216,19 +226,25 @@ class NeighborSampler {
 
 // Homogeneous neighbor sampling ///////////////////////////////////////////////
 
-template <bool replace, bool directed, bool disjoint, bool return_edge_id>
+template <bool replace,
+          bool directed,
+          bool disjoint,
+          bool return_edge_id,
+          bool distributed>
 std::tuple<at::Tensor,
            at::Tensor,
            at::Tensor,
            c10::optional<at::Tensor>,
            std::vector<int64_t>,
+           std::vector<int64_t>,
            std::vector<int64_t>>
 sample(const at::Tensor& rowptr,
        const at::Tensor& col,
        const at::Tensor& seed,
        const std::vector<int64_t>& num_neighbors,
        const c10::optional<at::Tensor>& time,
        const c10::optional<at::Tensor>& seed_time,
+       const c10::optional<at::Tensor>& batch,
```

Comment:
How does `batch` behave in case of `disjoint=False`? If distributed sampling requires `disjoint=True` anyway, I am not totally sure I understand why we need this new argument here.

Reply:
Distributed sampling does not requires `disjoint=true`. It can work with `disjoint=false` as well.
`batch` is used only when `disjoint=true`, otherwise it is not relevant.
Why `batch` is needed:
During distributed sampling we sample by one hop in c++ and go out of the sample() function. So if we sample more than one layer, information about which subgraph a given node belonged to, will be lost. So, thanks to the `batch` variable, we can assign initial values.